packages/dataviews/src/dataform-controls/password.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	privateApis,
+	__experimentalInputControlSuffixWrapper as InputControlSuffixWrapper,
+} from '@wordpress/components';
+import { useCallback, useState } from '@wordpress/element';
+import { seen, unseen } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import type { DataFormControlProps } from '../types';
+import { unlock } from '../lock-unlock';
+
+const { ValidatedInputControl } = unlock( privateApis );
+
+export default function Password< Item >( {
+	data,
+	field,
+	onChange,
+	hideLabelFromVision,
+}: DataFormControlProps< Item > ) {
+	const { id, label, placeholder, description } = field;
+	const value = field.getValue( { item: data } );
+	const [ isVisible, setIsVisible ] = useState( false );
+	const [ customValidity, setCustomValidity ] =
+		useState<
+			React.ComponentProps<
+				typeof ValidatedInputControl
+			>[ 'customValidity' ]
+		>( undefined );
+
+	const onChangeControl = useCallback(
+		( newValue: string ) =>
+			onChange( {
+				[ id ]: newValue,
+			} ),
+		[ id, onChange ]
+	);
+
+	const toggleVisibility = useCallback( () => {
+		setIsVisible( ( prev ) => ! prev );
+	}, [] );
+
+	return (
+		<ValidatedInputControl
```

Comment:
Can this use the [`utility/validated-text.tsx` component](https://github.com/WordPress/gutenberg/blob/461c970f4bb844cf01d4e5b561ad507dcda2f833/packages/dataviews/src/dataform-controls/utils/validated-text.tsx#L57) instead? I understand it just needs to accept a suffix. It'd be great if the utility was renamed to `utils/validated-input.tsx` as well (we forgot to that recently).